src/library/library.h

```diff
@@ -1,80 +1,76 @@
-// library.h
-// Created 8/23/2009 by RJ Ryan (rryan@mit.edu)
-
-// A Library class is a container for all the model-side aspects of the library.
-// A library widget can be attached to the Library object by calling bindLibraryWidget.
-
-#ifndef LIBRARY_H
-#define LIBRARY_H
+#pragma once
 
 #include <QList>
 #include <QObject>
 #include <QAbstractItemModel>
 #include <QFont>
+#include <QPointer>
 
+#include "analyzer/analyzerprogress.h"
 #include "preferences/usersettings.h"
-#include "track/globaltrackcache.h"
-#include "recording/recordingmanager.h"
-#include "library/analysisfeature.h"
-#include "library/coverartcache.h"
-#include "library/scanner/libraryscanner.h"
+#include "track/track.h"
 #include "util/db/dbconnectionpool.h"
 
-class TrackModel;
-class TrackCollection;
-class SidebarModel;
-class LibraryFeature;
-class LibraryTableModel;
-class WLibrarySidebar;
-class WLibrary;
-class MixxxLibraryFeature;
-class PlaylistFeature;
+class AnalysisFeature;
+class ControlObject;
 class CrateFeature;
+class ExternalTrackCollection;
 class LibraryControl;
+class LibraryFeature;
+class LibraryTableModel;
 class KeyboardEventFilter;
+class MixxxLibraryFeature;
 class PlayerManager;
+class PlaylistFeature;
+class RecordingManager;
+class SidebarModel;
+class TrackCollection;
+class TrackCollectionManager;
+class TrackModel;
+class WLibrarySidebar;
+class WLibrary;
 
-class ExternalTrackCollection;
-
-class Library: public QObject,
-    public virtual /*implements*/ GlobalTrackCacheSaver {
+// A Library class is a container for all the model-side aspects of the library.
+// A library widget can be attached to the Library object by calling bindLibraryWidget.
+class Library: public QObject {
     Q_OBJECT
 
   public:
     static const QString kConfigGroup;
 
-    static const ConfigKey kConfigKeyRepairDatabaseOnNextRestart;
-
     Library(QObject* parent,
             UserSettingsPointer pConfig,
             mixxx::DbConnectionPoolPtr pDbConnectionPool,
+            TrackCollectionManager* pTrackCollectionManager,
             PlayerManager* pPlayerManager,
             RecordingManager* pRecordingManager);
     ~Library() override;
 
-    void stopFeatures();
+    void stopPendingTasks();
 
-    mixxx::DbConnectionPoolPtr dbConnectionPool() const {
+    const mixxx::DbConnectionPoolPtr& dbConnectionPool() const {
         return m_pDbConnectionPool;
     }
 
-    TrackCollection& trackCollection() {
-        DEBUG_ASSERT(m_pTrackCollection);
-        return *m_pTrackCollection;
+    const QPointer<TrackCollectionManager>& trackCollections() const {
```

Comment:
Why return a reference to a QPointer instead of just a QPointer or a reference?

Reply:
Agreed. QPointer is recommended for storing it as a member variable, but not for passing around.